docs/python/readme/developing_callouts.rst

```diff
@@ -0,0 +1,140 @@
+.. _developing_callouts:
+
+Developing Callouts
+===================
+
+This repository provides the following files to be extended:
+
+* :ref:`CalloutServer <callout_server>`: Baseline service callout server.
+* :ref:`CalloutTools <callout_tools>`: Common functions for callout servers.
+
+Creating a New Server
+---------------------
+
+1. **Create a New Server Script**:
+
+    Create ``server.py`` and import the ``CalloutServer`` class:
+
+    .. code-block:: python
+
+        from extproc.service.callout_server import CalloutServer
+
+    Just from importing the server class we can make the server run by creating a new instance and calling the blocking function ``run``:
+
+    .. code-block:: python
+
+        if __name__ == '__main__':
+            CalloutServer().run()
+
+2. **Extend ``CalloutServer``**:
+
+    Calling the server like this won't do much besides respond to health checks. For the server to respond to callouts, we create a custom class extending ``CalloutServer``.
+
+    Make a class extending ``CalloutServer``:
+
+    .. code-block:: python
+
+        class BasicCalloutServer(CalloutServer):
+            def on_response_headers(self, headers: HttpHeaders, context: ServicerContext) -> HeadersResponse:
+                ...
+
+    There are a few callback methods in ``CalloutServer`` provided for developers to override:
+
+    * ``on_request_headers``: Process request headers.
+    * ``on_response_headers``: Process response headers.
+    * ``on_request_body``: Process request body.
+    * ``on_response_body``: Process response body.
+
+    These functions correspond to the ``oneof`` required field in a `ProcessingRequest <https://www.envoyproxy.io/docs/envoy/latest/api-v3/service/ext_proc/v3/external_processor.proto#service-ext-proc-v3-processingrequest>_` and required response field of a `ProcessingResponse <https://www.envoyproxy.io/docs/envoy/latest/api-v3/service/ext_procv3_processingresponse>_`.
```

Comment:
These links appear to be broken on my end.

Reply:
Both are working fine here on my end, could you check again with the new build?

Reply:
I think that the `` _` `` needs to be `` `_ `` Like below.

Reply:
Fixed, now it should be good, can you check that again to see if it is working now?